Survey/Survey/Sources/Presentation/Modules/Login/LoginView.swift

```diff
@@ -50,14 +48,34 @@ struct LoginView: View {
                     )
                     .accessibilityIdentifier("loginButtonIdentifier")
                 }
-                .alert(isPresented: $isLoginFailed) { () -> Alert in
-                    Alert(
-                        title: Text(Localize.login_failed_title()),
-                        message: Text(Localize.login_failed_description())
-                    )
-                }
-                .offset(y: editingMode ? -140 : -60)
+                .offset(y: editingMode ? -140.0 : -60.0)
                 .opacity(loginViewOpacity)
+                .disabled(viewModel.isLoading)
+
+                if viewModel.isLoginAttemptFailed {
+                    ZStack {
+                        BlurView()
+                    }
+                    .alert(isPresented: $shouldShowLoginFailedAlert) { () -> Alert in
+                        Alert(
+                            title: Text(Localize.login_failed_title()),
+                            message: Text(viewModel.errorMessage),
+                            dismissButton: .default(Text("OK")) {
+                                shouldShowLoginFailedAlert = true
+                                viewModel.didDismissLoginFailedAlert()
+                            }
+                        )
+                    }
+                }
```

Comment:
I think this is fine, just a little bit redundant to me when adding the `dismissButton` on `LoginView` and the `didDismissLoginFailedAlert()` on the `LoginViewModel`. We can achieve the same without adding those. Just a suggestion, feel free to ignore this 🙏 
```
.onReceive(viewModel.$isLoginAttemptFailed) { shouldShowLoginFailedAlert = $0 }
.alert(isPresented: $shouldShowLoginFailedAlert) { () -> Alert in
    Alert(
        title: Text(Localize.login_failed_title()),
        message: Text(viewModel.errorMessage)
    )
}
```